Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -175,6 +230,79 @@ public override void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        // Deserialize the given method and return bool to indicate whether the deserialized result should be used.
+        private bool TryDeserializeMethod(MethodInfo method, IIonReader reader, IonType ionType, out object deserialized)
+        {
+            // A setter should have exactly one argument.
+            var parameters = method.GetParameters();
+            if (parameters.Length != 1)
+            {
+                throw new InvalidOperationException(
+                    "An [IonPropertySetter] annotated method should have exactly one argument " +
+                    $"but {method.Name} has {parameters.Length} arguments");
+            }
+
+            deserialized = ionSerializer.Deserialize(reader, parameters[0].ParameterType, ionType);
+
+            return true;
+        }
+        
+        // Deserialize the given property and return bool to indicate whether the deserialized result should be used.
+        private bool TryDeserializeProperty(
+            PropertyInfo property, IIonReader reader, IonType ionType, out object deserialized)
+        {
+            // We deserialize whether or not we ultimately use the result because values
+            // for some Ion types need to be consumed in order to advance the reader.
+            deserialized = ionSerializer.Deserialize(reader, property.PropertyType, ionType);
+            
+            if (IsReadOnlyProperty(property))
+            {
+                // property.SetValue() does not work with a readonly property.
+                // logic for handling deserializing readonly properties happens during field processing
+                // when we detect backing fields for the property.
+                return false;
+            }
+
+            return !options.IgnoreDefaults || deserialized != default;
```

Comment:
`==` won't always work, I don't think. I think you need to do this: https://www.google.com/search?client=firefox-b-e&q=c%23+how+to+check+if+value+is+a+default. Since this just got moved, feel free to address that in a subsequent PR.